packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -86,11 +97,13 @@ const updateLocalBootData = (
 export const BootDataProvider = ({
   children,
   app,
+  localBootData,
   getRedirectUri,
 }: BootDataProviderProps): ReactElement => {
   const queryClient = useQueryClient();
-  const [cachedBootData, setCachedBootData] =
-    useState<Partial<BootCacheData>>();
+  const [cachedBootData, setCachedBootData] = useState<Partial<BootCacheData>>(
+    () => localBootData || getLocalBootData(true),
```

Comment:
If we assume `localBootData` is always provided we don't need the second part. Ideally we will read the cache only once

Reply:
In the instance of webapp which the flicker doesn't happen anymore, it doesn't make the fetch in a much earlier time. But we can definitely do it as well when the webapp initializes.